### ISSUE TITLE
Unable to process stream response with _performSecureRequest

### ISSUE THREAD

**Issue by cvng** [ciaranj/node-oauth#228]
Hi there, and thanks for this great oAuth implementation.

However, i found myself unable to work with stream response with actual implementation, because `_performSecureRequest` wait for the **end / close** event before returning data.

A simple fix would be `passBackControl` immediately, when response is a stream, and let the user collect the chunk:

``` javascript
if ( response.headers["transfer-encoding"] == "chunked")
    passBackControl( response );
```

Hoping that helps :-)

**Comment by hhamilto** [comment]
Not sure that I agree with the fix suggested here, but this library does need the ability to deal with streamed responses. +1

**Comment by hhamilto** [comment]
After taking a longer peek at the source, it looks like omitting a callback makes `_performSecureRequest` return the request object, which you can then treat any way you want.

**Comment by cvng** [comment]
Yes I remember that was for the Twitter streaming API, but that was 2 years now, no PR was merged in the meanwhile?

**Comment by hhamilto** [comment]
@ciaranj Would you accept a PR that contained a new example of how to interact with Twitter's streaming API?
